test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -225,11 +222,209 @@ func (v *volumeModifyTestSuite) DefineTests(driver storageframework.TestDriver,
 		l.resource.Pvc = SetPVCVACName(ctx, l.resource.Pvc, newVAC.Name, f.ClientSet, setVACWaitPeriod)
 		gomega.Expect(l.resource.Pvc).NotTo(gomega.BeNil())
 
-		ginkgo.By("Waiting for modification to finish")
-		l.resource.Pvc = WaitForVolumeModification(ctx, l.resource.Pvc, f.ClientSet, modifyVolumeWaitPeriod)
+		ginkgo.By("Checking PVC status")
+		err = e2epv.WaitForPersistentVolumeClaimModified(ctx, f.ClientSet, l.resource.Pvc, modifyVolumeWaitPeriod)
+		framework.ExpectNoError(err, "While waiting for PVC to have expected VAC")
+	})
 
-		pvcConditions := l.resource.Pvc.Status.Conditions
-		gomega.Expect(pvcConditions).To(gomega.BeEmpty(), "PVC should not have conditions")
+	ginkgo.It("should not modify volume when target VAC not found", func(ctx context.Context) {
+		init(ctx, true /* volume created with VAC */)
+		ginkgo.DeferCleanup(cleanup)
+
+		ginkgo.By("Creating a pod with dynamically provisioned volume")
+		podConfig := e2epod.Config{
+			NS:            f.Namespace.Name,
+			PVCs:          []*v1.PersistentVolumeClaim{l.resource.Pvc},
+			SeLinuxLabel:  e2epod.GetLinuxLabel(),
+			NodeSelection: l.config.ClientNodeSelection,
+			ImageID:       e2epod.GetDefaultTestImageID(),
+		}
+		pod, err := e2epod.CreateSecPodWithNodeSelection(ctx, f.ClientSet, &podConfig, f.Timeouts.PodStart)
+		ginkgo.DeferCleanup(e2epod.DeletePodWithWait, f.ClientSet, pod)
+		framework.ExpectNoError(err, "While creating pod for modifying")
+
+		ginkgo.By("Modifying PVC via non-existent VAC")
+		l.resource.Pvc = SetPVCVACName(ctx, l.resource.Pvc, "fake-vac", f.ClientSet, setVACWaitPeriod)
+		gomega.Expect(l.resource.Pvc).NotTo(gomega.BeNil())
+
```

Comment:
Note, no `VolumeModifyFailed` event emitted. I believe this is an external-resizer bug.

I will raise an issue on external resizer and make checklist item to add a `VolumeModifyFailed` event here.

Error message in resizer container:

```
csi-resizer E0729 00:03:26.778429       1 modify_volume.go:91] Get VAC with vac name <VAC-NAME> in VACInformer cache failed: volumeattributesclass.storage.k8s.io "io6-class" not found
```

Reply:
x-ref: https://github.com/kubernetes-csi/external-resizer/issues/427